pages/signup/index.js

```diff
@@ -1,126 +1,262 @@
 import React, { useState, useEffect } from 'react'
 import Image from 'next/image'
 import { useDispatch, useSelector } from 'react-redux'
-import { useRouter } from 'next/router'
-import { changeLoginType, signup } from 'redux/reducers/authSlice'
+import Router from 'next/router'
+import {} from 'redux/reducers/authSlice'
+import { getSession } from 'next-auth/react'
+
+import { REGISTER_USER } from 'utils/mutations'
+import { useMutation } from '@apollo/client'
+import { MOON_HOLDINGS } from 'app/constants/copy'
+import BannerModal from 'components/modals/BannerModal'
 
 const Index = () => {
-  const router = useRouter()
   const dispatch = useDispatch()
   const [email, setEmail] = useState('')
   const [password, setPassword] = useState('')
   const [confirmPassword, setConfirmPassword] = useState('')
+  const [message, setMessage] = useState('')
+  const [error, setError] = useState(false)
+  const [showModal, setShowModal] = useState(false)
+
+  const [signUp, { loading, data }] = useMutation(REGISTER_USER)
+
+  useEffect(() => {
+    if (data) {
+      setModal(
+        'You have successfully signed up. Please verify your email to login.',
+        false,
+        true
+      )
+      const timer = setTimeout(() => {
+        Router.push('/login')
+      }, 5000)
+      return () => {
+        clearTimeout(timer)
+      }
+    }
+  }, [data, dispatch])
 
-  const { signUpSuccess } = useSelector((state) => state.auth)
+  useEffect(() => {
+    if (showModal) {
+      const timer = setTimeout(() => {
+        setModal('', false, false)
+      }, 2000)
+      return () => {
+        clearTimeout(timer)
+      }
+    }
+  }, [showModal])
 
-  const register = () => {
-    if (password === confirmPassword) {
-      dispatch(signup({ email, password }))
+  const register = async () => {
+    if (
+      email.length == 0 ||
+      password.length == 0 ||
+      confirmPassword.length == 0
+    ) {
+      setModal('Please fill up all fields', true, true)
+    } else {
+      try {
+        await signUp({
+          variables: { email: email, password: password },
+        })
+      } catch (error) {
+        setModal(error.message, true, true)
+      }
     }
   }
 
-  const twitterAuth = () => {
-    dispatch(changeLoginType('twitter'))
-    window.open(
-      `${process.env.NEXT_PUBLIC_MOON_SERVER_URL}/api/auth/twitter`,
-      '_self'
+  const setModal = (message, error, show) => {
+    setShowModal(show)
+    setMessage(message)
+    setTimeout(
```

Comment:
Why need timeout? @micohalvarez

Reply:
to match the animation time of the modal when hiding. if no timeout the color change will be visible